openapi_python_client/parser/openapi.py

```diff
@@ -258,37 +279,38 @@ def _add_parameters(
             if isinstance(prop, ParseError):
                 return ParseError(detail=f"cannot parse parameter of endpoint {endpoint.name}", data=prop.data), schemas
 
-            if prop.python_name in used_python_names:
-                duplicate, duplicate_location = used_python_names[prop.python_name]
-                if duplicate.python_name == prop.python_name:  # Existing should be converted too for consistency
-                    duplicate.set_python_name(f"{duplicate.python_name}_{duplicate_location}", config=config)
-                prop.set_python_name(f"{prop.python_name}_{param.param_in}", config=config)
-            else:
-                used_python_names[prop.python_name] = (prop, param.param_in)
+            for location, parameters_dict in parameters_by_location.items():
+                existing_prop: Optional[Property] = parameters_dict.get(prop.name)
+                same_location: bool = location == param.param_in
+
+                if existing_prop and not same_location:
```

Comment:
```suggestion
                if location == param.param_in or prop.name not in parameters_dict:
                    continue
                existing_prop: Property = parameters_dict[prop.name]
```
This saves a bit of processing time when checking the same location and lets you tab out the following logic, reducing cognitive complexity.